input/kafka_avro.go

```diff
@@ -0,0 +1,137 @@
+package input
+
+import (
+	"context"
+	"encoding/binary"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/housepower/clickhouse_sinker/config"
+	"github.com/housepower/clickhouse_sinker/model"
+	"github.com/housepower/clickhouse_sinker/statistics"
+	"github.com/housepower/clickhouse_sinker/util"
+	"github.com/pkg/errors"
+	"github.com/riferrei/srclient"
+	"go.uber.org/zap"
+	"io"
+)
+
+var _ Inputer = (*KafkaAvro)(nil)
+
+// KafkaAvro implements input.Inputer
+type KafkaAvro struct {
+	consumer             *kafka.Consumer
+	cfg                  *config.Config
+	taskCfg              *config.TaskConfig
+	stopped              chan struct{}
+	putFn                func(msg model.InputMessage)
+	schemaRegistryClient *srclient.SchemaRegistryClient
+}
+
+// NewKafkaAvro get instance of kafka reader
+func NewKafkaAvro() *KafkaAvro {
+	return &KafkaAvro{}
+}
+
+// Init Initialise the kafka instance with configuration
+func (k *KafkaAvro) Init(cfg *config.Config, taskCfg *config.TaskConfig, putFn func(msg model.InputMessage), cleanupFn func()) (err error) {
+	k.cfg = cfg
+	k.taskCfg = taskCfg
+	k.stopped = make(chan struct{})
+	k.putFn = putFn
+	kfkCfg := &cfg.Kafka
+
+	offset := kafka.OffsetEnd
+	if k.taskCfg.Earliest {
+		offset = kafka.OffsetBeginning
+	}
+
+	k.consumer, err = kafka.NewConsumer(&kafka.ConfigMap{
+		"bootstrap.servers":  kfkCfg.Brokers,
+		"security.protocol":  kfkCfg.Security["protocol"],
```

Comment:
Avoid using Security and Sasl at the same time. It may confuse other users.

Reply:
How bad can performance be Avro+Json ? And how can this be avoided ?

Reply:
How bad can performance be Avro+Json ? And how can this be avoided ?